glommio/src/error.rs

```diff
@@ -47,6 +98,17 @@ mod test {
     use super::*;
     use std::io;
 
+    #[test]
```

Comment:
I would like to see tests making sure that we can convert easily to io::Error too.

For example, unless I am missing something it doesn't seem to me that this is providing a translation from the semaphore closed error to an io::Error / BrokenPipe.

Reply:
Oh -- do you need that? I assumed it was just a made up error picked in that case for convenience. Largely you are only meant to go from io::Error -> GlommioError but it'll also be the cause of an error so it could be pulled out there. But is there need to go BACK to io::Error types?

Reply:
I think it is fundamental to be able to convert from glommio errors to io::Errors.
As I said before my main oposition to even having our own error is that every single other thing out there that deals with I/O (network and storage) tends to use `io::Error`. If we can't convert automatically, you lose the ability to do `?`, and the implementation of `From` usually needs to be crate local.

Once we accept that we have to implement `From` to convert between glommio errors to `io::Error` for interoperability, then there are two choices: we either stash everything into `io::Error::Other`, or we try our best to keep meaningful error kinds whenever possible.

I heavily prefer the latter, because it is easier for users to figure out what is going on.

Nothing of this is set of stone: if you see it differently by all means voice it.
I want to make it clear that the reason I am pushing for this is the use case where we are using, let's say, `hyper` with `glommio`: `hyper` uses `io::Error` everywhere. And not only that, but they are certainly testing for certain classes of error to see if a socket is disconnected, etc. If we are stashing everything into `Other`, aren't we making the integration harder?